content/en/examples/controllers/job-pod-failure-policy-config-issue.yaml

```diff
@@ -17,3 +17,4 @@ spec:
     - action: FailJob
       onPodConditions:
       - type: ConfigIssue
+        status: "True"
```

Comment:
Interesting, can you please double check it really raises validation errors for you? 

I just checked `kubectl create -f job-pod-failure-policy-config-issue.yaml` and it passed for me. 

We have defaulting code to make sure the status=True if not specified: https://github.com/kubernetes/kubernetes/blob/56b5f05aea6f32db3b81417ce5d3508dc80a33ea/pkg/apis/batch/v1/defaults.go#L59.

Reply:
Hmm.  I did get a validation error on a dev build.  That was why I added this.  

I did not have PodFailurePolicy or PodDisruptionCondition set when I ran `local-up-cluster.sh` but it should be activated already.

I'll look into this AM.

Reply:
Thanks. This example requires `JobPodFailurePolicy` and `PodDisruptionConditions` enabled. Also, you may want to test master in case you have some changes on your dev branch.

Reply:
I ran:

```
apiVersion: batch/v1
kind: Job
metadata:
  name: job-pod-failure-policy-example
spec:
  completions: 12
  parallelism: 3
  template:
    spec:
      restartPolicy: Never
      containers:
      - name: main
        image: docker.io/library/bash:5
        command: ["bash"]        
        args:
        - -c
        - echo "Hello world!" && sleep 5 && exit 42
  backoffLimit: 6
  podFailurePolicy:
    rules:
    - action: Ignore
      onPodConditions:
      - type: DisruptionTarget
```

I used latest master (`sudo PATH=$PATH FEATURE_GATES=JobPodFailurePolicy=true PodDisruptionCondition=true hack/local-up-cluster.sh`)

And when I run this I get:

```
[ec2-user@ip-172-31-93-184 Job-Slow-Cleanup]$ kubectl create -f job-example-pod-failure-policy.yaml 
error: error validating "job-example-pod-failure-policy.yaml": error validating data: ValidationError(Job.spec.podFailurePolicy.rules[0].onPodConditions[0]): missing required field "status" in io.k8s.api.batch.v1.PodFailurePolicyOnPodConditionsPattern;
```

I also verify that these fields are stored in the api-server so the feature is turned on.

Reply:
Weird, but in that case it seems like a bug rather than something to update docs. Could you also test on a released version, such as 1.26 or 1.27? I think we need to investigate why the defaulting does not kick in.

Reply:
Hm, but after you start the cluster using `hack/local-up-cluster.sh` can you interact with it just with `kubectl`. When I repeated the steps I got the info at the end:

```sh
To start using your cluster, you can open up another terminal/tab and run:

  export KUBECONFIG=/var/run/kubernetes/admin.kubeconfig
  cluster/kubectl.sh

Alternatively, you can write to the default kubeconfig:

  export KUBERNETES_PROVIDER=local

  cluster/kubectl.sh config set-cluster local --server=https://localhost:6443 --certificate-authority=/var/run/kubernetes/server-ca.crt
  cluster/kubectl.sh config set-credentials myself --client-key=/var/run/kubernetes/client-admin.key --client-certificate=/var/run/kubernetes/client-admin.crt
  cluster/kubectl.sh config set-context local --cluster=local --user=myself
  cluster/kubectl.sh config use-context local
  cluster/kubectl.sh
```

So I did in a different shell: 
```sh
export KUBECONFIG=/var/run/kubernetes/admin.kubeconfig
./cluster/kubectl.sh create -f ../website/content/en/examples/controllers/job-pod-failure-policy-config-issue.yaml 
Using image: cos-97-16919-294-51 from project: cos-cloud as master image
Using image: cos-97-16919-294-51 from project: cos-cloud as node image
job.batch/job-pod-failure-policy-config-issue created
```
and the job is ok:
```
...
    podFailurePolicy:
      rules:
      - action: FailJob
        onExitCodes: null
        onPodConditions:
        - status: "True"
          type: ConfigIssue
...          
```
I have also confirmed it works ok on 1.27.3 on kind.